pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -164,16 +166,21 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcileResult, err
 
 	} else if engine.Instance.Spec.EngineState == "active" {
-		r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "Started reconcile for chaosEngine", "Will create all Chaos resources for chaosUID: %v", string(engine.Instance.UID))
-		err := r.addFinalzerToEngine(engine, "chaosengine.litmuschaos.io/finalizer")
+
+		err := r.updateStatus(engine, "initialized")
 		if err != nil {
-			reqLogger.Error(err, "Unable to add Finalizers in ChaosEngine Resource, due to error: %v", err)
+			reqLogger.Error(err, "Unable to Update Status in ChaosEngine Resource, due to error: %v", err)
 		}
-		// Patch Status to initialized
-		err = r.updateStatus(engine, "initialized")
+
+		err = r.addFinalzerToEngine(engine, finalizer)
 		if err != nil {
-			reqLogger.Error(err, "Unable to Update Status in ChaosEngine Resource, due to error: %v", err)
+			reqLogger.Error(err, "Unable to add Finalizers in ChaosEngine Resource, due to error: %v", err)
 		}
+
+	} else if engine.Instance.Spec.EngineState == "" {
```

Comment:
I think this check need to place before checking the status as active

Reply:
It's an else if, any sequence would work.

Reply:
It should come as a separate if condition

Reply:
Otherwise, it will miss the edge case!!
Let say some new engine is applied with empty state then firstly we have to update his status as active instead of checking anything otherwise in first reconcile it will not update the status as initialized also it will not add the finalizer maybe it will do these step in next reconcile wich is not good!!

Reply:
agreed

Reply:
Many reconciles occur at a single point (due to watcher watching so many resources), so this case is already happening, still will improve in the PR.